channeldb/graph.go

```diff
@@ -216,22 +217,132 @@ func NewChannelGraph(db kvdb.Backend, rejectCacheSize, chanCacheSize int,
 		startTime := time.Now()
 		log.Debugf("Populating in-memory channel graph, this might " +
 			"take a while...")
+
 		err := g.ForEachNodeCacheable(
 			func(tx kvdb.RTx, node GraphCacheNode) error {
-				return g.graphCache.AddNode(tx, node)
+				g.graphCache.AddNodeFeatures(node)
+
+				return nil
 			},
 		)
 		if err != nil {
 			return nil, err
 		}
 
+		err = g.ForEachChannel(func(info *ChannelEdgeInfo,
+			policy1, policy2 *ChannelEdgePolicy) error {
+
+			g.graphCache.AddChannel(info, policy1, policy2)
+
+			return nil
+		})
+		if err != nil {
+			return nil, err
+		}
+
 		log.Debugf("Finished populating in-memory channel graph (took "+
 			"%v, %s)", time.Since(startTime), g.graphCache.Stats())
 	}
 
 	return g, nil
 }
 
+type channelMapKey struct {
+	nodeKey route.Vertex
+	chanID  [8]byte
+}
+
+func (c *ChannelGraph) forEachChannelFromMap(
+	edgeIndex kvdb.RBucket,
+	channelMap map[channelMapKey]*ChannelEdgePolicy,
```

Comment:
Instead of passing in the map, we could call `getChannelMap` here directly since it's not used anywhere else from what I can see. Or is the order important here (e.g. first load edges before trying to access the edge index bucket, related to my previous comment about what error is returned?).

Reply:
This is a left over from a previous revision. Currently the `forEachChannelFromMap` can just as well be integrated in `ForEachChannel`. Did that.